DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -389,7 +438,225 @@ void ZCounting::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+        const auto ele = electrons->ptrAt(iele);
+        if(*sc == *(ele->superCluster())) {
+          eleProbe = ele;
+          break;
+        }
+      }
+
+      // Assign final probe 4-vector
+      if(eleProbe.isNonnull()){
+        vProbe.SetPtEtaPhiM( eleProbe->pt(), eleProbe->eta(), eleProbe->phi(), ELECTRON_MASS);
+      } else {
+        double pt = sc->energy() * TMath::Sqrt( 1 - pow(TMath::TanH(sc->eta()),2) );
+        vProbe.SetPtEtaPhiM( pt, sc->eta(), sc->phi(), ELECTRON_MASS);
+      }
+
+      // Probe Selection
+      double probe_pt = vProbe.Pt();
+      double probe_abseta = fabs(sc->eta());
+      bool probe_is_valid_probe = ele_probe_selection(probe_pt, probe_abseta);
+      if( !probe_is_valid_probe ) continue;
+
+      // Good Probe found!
+
+      // Require good Z
+      vDilep = vTag + vProbe;
+      float MASS_LOW = 80.0;
+      float MASS_HIGH = 100.0;
+      if((vDilep.M()<MASS_LOW) || (vDilep.M()>MASS_HIGH)) continue;
+      if(eleProbe.isNonnull() and (eleProbe->charge() != - el1->charge())) continue;
+
+      // Good Z found!
+
+      long ls = iEvent.luminosityBlock();
+      bool probe_pass_trigger = isElectronTriggerObj(*fTrigger, TriggerTools::matchHLT(vProbe.Eta(), vProbe.Phi(), fTrigger->fRecords, *hTrgEvt));
+      bool probe_pass_id = eleProbe.isNonnull() and EleID_.passID(eleProbe);
+
+      //// Fill for yields
+      bool probe_is_forward = probe_abseta > ELE_ETA_CRACK_LOW;
+      bool tag_is_forward = tag_abseta > ELE_ETA_CRACK_LOW;
+
+      if(probe_pass_id) {
+        if(probe_is_forward and tag_is_forward) {
+          h_ee_yield_Z_eeee -> Fill(ls);
+        } else if(!probe_is_forward and !tag_is_forward) {
+          h_ee_yield_Z_ebeb -> Fill(ls);
+        } else {
+          h_ee_yield_Z_ebee -> Fill(ls);
+        }
+      }
+
+      if(!tag_is_valid_tag) continue;
+
+      /// Fill for ID efficiency
+      if(probe_pass_id) {
+        if(probe_is_forward) {
+          h_ee_mass_id_pass_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_id_pass_central->Fill(ls, vDilep.M());
+        }
+      } else {
+          if(probe_is_forward) {
+            h_ee_mass_id_fail_forward->Fill(ls, vDilep.M());
+          } else {
+            h_ee_mass_id_fail_central->Fill(ls, vDilep.M());
+          }
+      }
+
+      /// Fill for HLT efficiency
+      if(probe_pass_id and probe_pass_trigger) {
+        if(probe_is_forward) {
+          h_ee_mass_HLT_pass_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_HLT_pass_central->Fill(ls, vDilep.M());
+        }
+      } else if (probe_pass_id) {
+        if(probe_is_forward) {
+          h_ee_mass_HLT_fail_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_HLT_fail_central->Fill(ls, vDilep.M());
+        }
+      }
+    } // End of probe loop
+  }//End of tag loop
+
+}
+bool ZCounting::ele_probe_selection(double pt, double abseta){
+  bool pass = true;
+  pass &= pt > ELE_PT_CUT_PROBE;
```

Comment:
these functions can be more efficiently written as
if (pt<= ELE_PT_CUT_PROBE) return false;
...
return true;

as once pass is false, the function execution can be ended.

Reply:
Done.